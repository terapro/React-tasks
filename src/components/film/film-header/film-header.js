import React from 'react';
import PropTypes from 'prop-types';

import {Title} from 'src/components/common/title/title.js';
import {Rating} from 'src/components/common/rating/rating.js';


const FilmHeader =(props) => {

  return (
    <div className='film-header' >
      <div>
        <Title content={props.title} ttlRed={true} ttlBig={true} />
      </div>
      <div>
        <Rating value ={props.rating}/>
      </div>
    </div>
  );

};

FilmHeader.propTypes = {
  title: PropTypes.string,
  rating: PropTypes.number
};


export {FilmHeader};