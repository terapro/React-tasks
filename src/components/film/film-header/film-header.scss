.film-header {
  display: flex;
  align-items: end;
  cursor: default;
}