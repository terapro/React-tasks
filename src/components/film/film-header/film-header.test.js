import React from 'react';
import {FilmHeader} from 'src/components/film/film-header/film-header.js';
import {shallow} from 'enzyme';

const {title, rating} = {
  title: 'film title',
  rating: 4.2
};

describe('<FilmHeader />', () => {
  describe('rendering', () => {
    it('should render the component', () => {
      const wrapper = shallow(<FilmHeader title={title} rating={rating} />);
      expect(wrapper).toMatchSnapshot();
    });
    it('should render the component with default props', () => {
      const wrapper = shallow(<FilmHeader />);
      expect(wrapper).toMatchSnapshot();
    });
  });

});