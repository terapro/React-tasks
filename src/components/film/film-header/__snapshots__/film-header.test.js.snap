// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmHeader /> <FilmHeader /> should render the component 1`] = `
<div
  className="film-header"
>
  <div>
    <Title
      content="film title"
      ttlBig={true}
      ttlRed={true}
      ttlUC={false}
      ttlWhite={false}
    />
  </div>
  <div>
    <Rating
      value={4.2}
    />
  </div>
</div>
`;

exports[`<FilmHeader /> <FilmHeader /> should render the component with default props 1`] = `
<div
  className="film-header"
>
  <div>
    <Title
      content="Film title"
      ttlBig={true}
      ttlRed={true}
      ttlUC={false}
      ttlWhite={false}
    />
  </div>
  <div>
    <Rating
      value={0}
    />
  </div>
</div>
`;

exports[`<FilmHeader /> rendering should render the component 1`] = `
<div
  className="film-header"
>
  <div>
    <Title
      content="film title"
      ttlBig={true}
      ttlRed={true}
      ttlUC={false}
      ttlWhite={false}
    />
  </div>
  <div>
    <Rating
      value={4.2}
    />
  </div>
</div>
`;

exports[`<FilmHeader /> rendering should render the component with default props 1`] = `
<div
  className="film-header"
>
  <div>
    <Title
      content="Film title"
      ttlBig={true}
      ttlRed={true}
      ttlUC={false}
      ttlWhite={false}
    />
  </div>
  <div>
    <Rating
      value={0}
    />
  </div>
</div>
`;

exports[`<FilmHeader /> should render the component 1`] = `
<div
  className="film-header"
>
  <div>
    <Title
      content="film title"
      ttlBig={true}
      ttlRed={true}
      ttlUC={false}
      ttlWhite={false}
    />
  </div>
  <div>
    <Rating
      value={4.2}
    />
  </div>
</div>
`;
